src/paging.md

```diff
@@ -128,6 +128,14 @@ So here’s the strategy: create a single entry of each of these tables, then
 point them at each other in the correct way, then tell the CPU that paging
 should be enabled.
 
+### How many tables?
+
+The number of tables we need depends on how big we make each page. The bigger
+each page, the fewer pages fit into the virtual address space, so the fewer
+tables we need. How to choose a page size is the kind of detail we don't need to
+worry about for now. We're just going to go for 2 MiB pages, which means we only
+need three tables: we won't need a level 1 page table.
```

Comment:
IIRC, since x86 has paging hardware, we're also limited by what it supports. There's pages and huge pages. The reason why those sizes were chosen is important though, and if you were on a platform that didn't support it in hardware, you could do whatever.

I have no idea if any of that info belongs in this block.

Reply:
These days, on `x86_64`, you can choose between three page sizes: 4 KiB, 2 MiB and 1 GiB. The default is 4 KiB. If you set the page size flag (a.k.a. the "huge page" bit) in a P2 table entry, then that means that the entry points directly at a 2 MiB page instead of at a P1 table. Likewise, if you set the page size flag in a P3 table entry, that means that the entry points directly at a 1 GiB page instead of at a P2 table.

The thing that confused me when I was following this tutorial was that in the previous paragraph you list the four layers of page tables... and then we only use three of them. I wondered what had happened to `p1_table`. Then I discovered that you don't use a `p1_table` if all your pages are 2 MiB. So I wanted to add an explanation for the absence of `p1_table`.

I don't think we need to explain everything here: in general, you've been giving just enough information to make progress, without getting too bogged down in all the details. I think that works really well. Explaining all the trade-offs between different page sizes here would be a massive digression.